### ISSUE TITLE
Confusing behaviour in NavigationMiddlewareFactory

### ISSUE THREAD

**Issue by ghost** [froschdesign/zend-expressive-navigation#5]
When I first used this library I removed the following code from NavigationMiddlewareFactory:

```
        if (\count($names) === 1) {
            $this->containerNames[] = Navigation::class;
            return $this->containerNames;
        }
```

Currently, instead, I have a second, empty, navigation array in my navigation configuration to avoid triggering it. I'm not really sure of its purpose to be honest! If I only have one navigation declared - let's call it "main", it adds "Zend\Navigation\Navigation" to the list of containers, rather than "Zend\Navigation\Main", which would be added by the code that follows it:
```
        foreach ($names as $name) {
            $this->containerNames[] = self::SERVICE_PREFIX . ucfirst($name);
        }
```

I thought possibly it was a mistake and it was intended to be "count($names) === 0" to add "Zend\Navigation\Navigation" as a default or something, but am unsure.

Please could you clarify?

**Comment by froschdesign** [comment]
@kgrey 
> When I first used this library I removed the following code from NavigationMiddlewareFactory:

First: You should never edit any file in the `vendor` directory. Create your own factory instead.

> If I only have one navigation declared - let's call it "main", it adds "Zend\Navigation\Navigation" to the list of containers, rather than "Zend\Navigation\Main"

If only one container configuration was found, then the name `Zend\Navigation\Navigation` is used. The idea based on the backward compatibility, but this ends up in too many problems. I have to rethink this.

Thanks for reporting!

**Comment by froschdesign** [comment]
Fixed with fca00168ceb398e93a638737670770ebde1123f1
